### PR TITLE
Add how to's for etcd, consul and cassandra

### LINKED ISSUE #215: Add documentation for HashiCorp Consul state store
Dapr now has support for HashiCorp [Consul](https://consul.io) as a statestore.

https://github.com/dapr/dapr/pull/686

Documentation on how to setup and query the state store should be added in the following places:
* https://github.com/dapr/docs/tree/master/howto/setup-state-store
* https://github.com/dapr/docs/tree/master/howto/setup-state-store

### PR DESCRIPTION
**Pull request by yaron2**
Adds How To tutorials for the following state stores:

* Cassandra
* etcd
* Consul
* Memcached

closes #215

### CODE REVIEW

File: howto/setup-state-store/setup-cassandra.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Setup Cassandra 
+
+## Locally
+
+You can run Cassandra locally with th Datastax Docker image:
```
Comment (lukekim) [inline howto/setup-state-store/setup-cassandra.md]: The

File: howto/setup-state-store/setup-cassandra.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+```
+docker run -e DS_LICENSE=accept --memory 4g --name my-dse -d datastax/dse-server -g -s -k
+```
+
+You can then interact with the server using `localhost:9042`.
```
Comment (lukekim) [inline howto/setup-state-store/setup-cassandra.md]: Should we use `127.0.0.1` here instead?

File: howto/setup-state-store/setup-cassandra.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+To apply the Cassandra state store to Kubernetes, use the `kubectl` CLI:
+
+```
+kubectl apply -f cassandra.yaml
+```
+
+### Running locally
+
+The Dapr CLI will automatically create a directory named `components` in your current working directory with a Redis component.
+To use Cassandra, replace the redis.yaml file with the cassandra.yaml above.
```
Comment (lukekim) [inline howto/setup-state-store/setup-cassandra.md]: `redis.yaml`? `cassandra.yaml`?

File: howto/setup-state-store/setup-consul.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+## Locally
+
+You can run Consul locally using Docker:
+
+```
+docker run -d --name=dev-consul -e CONSUL_BIND_INTERFACE=eth0 consul
+```
+
+You can then interact with the server using `localhost:8500`.
```
Comment (lukekim) [inline howto/setup-state-store/setup-consul.md]: Same comment as above.

File: howto/setup-state-store/setup-consul.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+To apply the Consul state store to Kubernetes, use the `kubectl` CLI:
+
+```
+kubectl apply -f consul.yaml
+```
+
+### Running locally
+
+The Dapr CLI will automatically create a directory named `components` in your current working directory with a Redis component.
+To use Consul, replace the redis.yaml file with the consul.yaml above.
```
Comment (lukekim) [inline howto/setup-state-store/setup-consul.md]: Same comment as above.

File: howto/setup-state-store/setup-etcd.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+## Locally
+
+You can run etcd locally using Docker:
+
+```
+docker run -d --name etcd bitnami/etcd
+```
+
+You can then interact with the server using `localhost:2379`.
```
Comment (lukekim) [inline howto/setup-state-store/setup-etcd.md]: `127.0.0.1`?

File: howto/setup-state-store/setup-cassandra.md (hunk 2 above)
Comment (yaron2) [reply howto/setup-state-store/setup-cassandra.md]: This is for docker, best to keep consistent with `localhost` as most samples show.

File: howto/setup-state-store/setup-cassandra.md (hunk 3 above)
Comment (yaron2) [reply howto/setup-state-store/setup-cassandra.md]: Yeah, it says to replace `redis.yaml` which is the default with `cassandra.yaml` created above.

File: howto/setup-state-store/setup-consul.md (hunk 4 above)
Comment (yaron2) [reply howto/setup-state-store/setup-consul.md]: Answered above

File: howto/setup-state-store/setup-etcd.md (hunk 6 above)
Comment (yaron2) [reply howto/setup-state-store/setup-etcd.md]: Answered above

File: howto/setup-state-store/setup-cassandra.md (hunk 1 above)
Comment (yaron2) [reply howto/setup-state-store/setup-cassandra.md]: Fixed

**Comment by yaron2** [comment]
Addressed review comments and added Memcached.

**Pull request merged**